cmd/node/main.go

```diff
@@ -634,32 +685,149 @@ func startNode(ctx *cli.Context, log *logger.Logger, version string) error {
+	pollingInterval int,
+	networkComponents *factory.Network,
+	processComponents *factory.Process,
+) error {
+
 	if ash == nil {
 		return errors.New("nil AppStatusHandler")
 	}
 
-	pollingIntervalDuration := time.Duration(pollingInterval) * time.Second
-	appStatusPollingHandler, err := appStatusPolling.NewAppStatusPolling(ash, pollingIntervalDuration)
+	appStatusPollingHandler, err := appStatusPolling.NewAppStatusPolling(ash, pollingInterval)
 	if err != nil {
 		return errors.New("cannot init AppStatusPolling")
 	}
 
+	err = registerPollConnectedPeers(appStatusPollingHandler, networkComponents)
+	if err != nil {
+		return err
+	}
+
+	err = registerPollProbableHighestNonce(appStatusPollingHandler, processComponents)
+	if err != nil {
+		return err
+	}
+
+	appStatusPollingHandler.Poll()
+
+	return nil
+}
+
+func registerPollConnectedPeers(
+	appStatusPollingHandler *appStatusPolling.AppStatusPolling,
+	networkComponents *factory.Network,
+) error {
+
 	numOfConnectedPeersHandlerFunc := func(appStatusHandler core.AppStatusHandler) {
-		numOfConnectedPeers := int64(len(networkComponents.NetMessenger.ConnectedAddresses()))
-		appStatusHandler.SetInt64Value(core.MetricNumConnectedPeers, numOfConnectedPeers)
+		numOfConnectedPeers := uint64(len(networkComponents.NetMessenger.ConnectedAddresses()))
+		appStatusHandler.SetUInt64Value(core.MetricNumConnectedPeers, numOfConnectedPeers)
 	}
-	err = appStatusPollingHandler.RegisterPollingFunc(numOfConnectedPeersHandlerFunc)
 
+	err := appStatusPollingHandler.RegisterPollingFunc(numOfConnectedPeersHandlerFunc)
 	if err != nil {
 		return errors.New("cannot register handler func for num of connected peers")
 	}
 
+	return nil
+}
+
+func registerPollProbableHighestNonce(
+	appStatusPollingHandler *appStatusPolling.AppStatusPolling,
+	processComponents *factory.Process,
+) error {
+
+	probableHighestNonceHandlerFunc := func(appStatusHandler core.AppStatusHandler) {
+		probableHigherNonce := processComponents.ForkDetector.ProbableHighestNonce()
+		appStatusHandler.SetUInt64Value(core.MetricProbableHighestNonce, probableHigherNonce)
+	}
+
+	err := appStatusPollingHandler.RegisterPollingFunc(probableHighestNonceHandlerFunc)
+	if err != nil {
+		return errors.New("cannot register handler func for forkdetector's probable higher nonce")
+	}
+
+	return nil
+}
+
+func startMachineStatisticsPolling(ash core.AppStatusHandler, pollingInterval int) error {
+	if ash == nil {
+		return errors.New("nil AppStatusHandler")
+	}
+
+	appStatusPollingHandler, err := appStatusPolling.NewAppStatusPolling(ash, pollingInterval)
+	if err != nil {
+		return errors.New("cannot init AppStatusPolling")
+	}
+
+	err = registerCpuStatistics(appStatusPollingHandler)
+	if err != nil {
+		return err
+	}
+
+	err = registerMemStatistics(appStatusPollingHandler)
+	if err != nil {
+		return err
+	}
+
+	err = registeNetStatistics(appStatusPollingHandler)
+	if err != nil {
+		return err
+	}
+
 	appStatusPollingHandler.Poll()
```

Comment:
The object `appStatusPollingHandler` will go out of scope at the end of the current function. But that's okay, because it has started all the necessary go routines, right?

Reply:
yes